src/caffe/proto/caffe.proto

```diff
@@ -445,6 +450,40 @@ message DataParameter {
   optional bool mirror = 6 [default = false];
 }
 
+// Message that stores parameters used by DataLayer
+message DatumDBParameter {
+  enum Mode {
+    NEW = 0;
+    WRITE = 1;
+    READ = 2;
+  }
+  // Specify the datumDB Backend.
+  optional string backend = 1 [default = "leveldb"];
```

Comment:
Initially I used `enum` but then changed thinking on #1685 and thinking in external extensions than don't require changing `caffe.proto`